src/Approximations/overapproximate.jl

```diff
@@ -1510,3 +1510,121 @@ function overapproximate(am::AbstractAffineMap{N, <:AbstractHyperrectangle{N}},
     radius_MX = abs.(M) * radius_hyperrectangle(X)
     return Hyperrectangle(center_MXv, radius_MX)
 end
+
+"""
+    overapproximate(X::LazySet{N}, ::Type{<:Zonotope},
+                    dir::AbstractDirections{N};
+                    solver=default_lp_solver(N)) where {N<:Real}
+
+Overapproximate a polytopic set with a zonotope of minimal total generator sum
+using only generators in the given directions.
+
+### Input
+
+- `X`        -- polytopic set
+- `Zonotope` -- type for dispatch
+- `dir`      -- directions used for the generators
+- `solver`   -- (optional, default: `default_lp_solver(N)`) the backend used to
+                solve the linear program
+
+### Output
+
+A zonotope that overapproximates `X` and uses at most the directions provided in
+`dir` (redundant directions will be ignored).
+Under this constraint, the zonotope has the minimal sum of generator vectors.
+
+### Notes
+
+The algorithm only requires one representative of each generator direction and
+their additive inverse (e.g. only one of `[1, 0]` and `[-1, 0]`) and assumes
+that the directions are normalized.
+We preprocess the directions in that respect.
+
+### Algorithm
+
+We solve a linear program parametric in the vertices ``v_j`` of `X` and the
+directions ``d_k`` in `dir` presented in [1, Section 4.2] (adapting the notation
+to the one used in this library).
+
+```math
+    \\min \\sum_{k=1}^l c_k \\
+    s.t. \\
+    \\text{center} + \\sum_{k=1}^l b_{kj} * d_k = v_j \\quad \\forall j \\
```

Comment:
can you use `c` for `\\text{center}` , in either case  define it below ?

Reply:
`c` conflicts with `c_k`.

Reply:
I renamed `c_k` to `α_k`.